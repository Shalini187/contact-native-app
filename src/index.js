import ScreenOne from './screen_1/screenone';
import ScreenTwo from './screen_2/screentwo';
import ScreenThree from './screen_3/screenthree';
import ScreenFour from './screen_4/screenfour';

export {ScreenOne};
export {ScreenTwo};
export {ScreenThree};
export {ScreenFour};