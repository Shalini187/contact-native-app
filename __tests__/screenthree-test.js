import 'react-native';
import React from 'react';

import {render, fireEvent} from 'react-native-testing-library';
import '@testing-library/jest-dom/extend-expect';

import {ScreenThree} from '../src';

test('renders Validation Check',  () => {
        const { getByTestId, toJSON } = render(<ScreenThree />);

        const email = getByTestId('email');
        const password = getByTestId('password');

        fireEvent.changeText(email, "");
        fireEvent.changeText(password, "");

        const input = getByTestId('Pressable');

        fireEvent.press(input);
        expect(toJSON()).toMatchSnapshot(); 
});


test('renders Validation Check',  () => {
        const { getByTestId, toJSON } = render(<ScreenThree />);

        const email = getByTestId('email');
        const password = getByTestId('password');

        fireEvent.changeText(email, "email123@gmail.com");
        fireEvent.changeText(password, "Email@123");

        const input = getByTestId('Pressable');

        fireEvent.press(input);
        expect(toJSON()).toMatchSnapshot(); 
});

test('renders Validation Check',  () => {
        const { getByTestId, toJSON } = render(<ScreenThree />);

        const email = getByTestId('email');
        const password = getByTestId('password');

        fireEvent.changeText(email, "email@123gmail.com");
        fireEvent.changeText(password, "email@123");

        const input = getByTestId('Pressable');

        fireEvent.press(input);
        expect(toJSON()).toMatchSnapshot(); 
});

test('renders Validation Check',  () => {
        const { getByTestId, toJSON } = render(<ScreenThree />);

        const email = getByTestId('email');
        const password = getByTestId('password');

        fireEvent.changeText(email, "email12gmail.com");
        fireEvent.changeText(password, "Email@12");

        const input = getByTestId('Pressable');

        fireEvent.press(input);
        expect(toJSON()).toMatchSnapshot(); 
});


