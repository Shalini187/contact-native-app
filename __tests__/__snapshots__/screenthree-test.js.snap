// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Validation Check 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "top": 40,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Image/image_3.jpg",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
        NUEGOV
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter your email address"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="email"
        title="Enter Email"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your password"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="password"
        title="Enter Password"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Forgot Password?
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(10,10,250,100)",
            "margin": 15,
            "padding": 10,
          }
        }
        testID="Pressable"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
           LOGIN 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`renders Validation Check 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "top": 40,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Image/image_3.jpg",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
        NUEGOV
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter your email address"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="email"
        title="Enter Email"
        underlineColorAndroid="transparent"
        value="email123@gmail.com"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your password"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="password"
        title="Enter Password"
        underlineColorAndroid="transparent"
        value="Email@123"
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Forgot Password?
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(10,10,250,100)",
            "margin": 15,
            "padding": 10,
          }
        }
        testID="Pressable"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
           LOGIN 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`renders Validation Check 3`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "top": 40,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Image/image_3.jpg",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
        NUEGOV
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter your email address"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="email"
        title="Enter Email"
        underlineColorAndroid="transparent"
        value="email@123gmail.com"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your password"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="password"
        title="Enter Password"
        underlineColorAndroid="transparent"
        value="email@123"
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Forgot Password?
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(10,10,250,100)",
            "margin": 15,
            "padding": 10,
          }
        }
        testID="Pressable"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
           LOGIN 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`renders Validation Check 4`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "top": 40,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Image/image_3.jpg",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
        NUEGOV
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter your email address"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="email"
        title="Enter Email"
        underlineColorAndroid="transparent"
        value="email12gmail.com"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your password"
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.10)",
            "color": "black",
            "height": 50,
            "margin": 15,
            "padding": 10,
            "textAlign": "center",
            "width": 300,
          }
        }
        testID="password"
        title="Enter Password"
        underlineColorAndroid="transparent"
        value="Email@12"
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Forgot Password?
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(10,10,250,100)",
            "margin": 15,
            "padding": 10,
          }
        }
        testID="Pressable"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
           LOGIN 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
